File: app/src/main/java/br/com/vicentec12/mygames/data/model/ConsoleWithGames.java
package br.com.vicentec12.mygames.data.model;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;
import androidx.room.Embedded;
import androidx.room.Relation;

import java.util.List;
import java.util.Objects;

public class ConsoleWithGames {

    @Embedded
    private Console console;

    @Relation(parentColumn = "id", entityColumn = "id_console")
    private List<Game> games;

    public Console getConsole() {
        return console;
    }

    public void setConsole(Console console) {
        this.console = console;
    }

    public List<Game> getGames() {
        return games;
    }

    public void setGames(List<Game> games) {
        this.games = games;
    }

    public static DiffUtil.ItemCallback<ConsoleWithGames> getDiffItemCallback() {
        return new DiffUtil.ItemCallback<ConsoleWithGames>() {
            @Override
            public boolean areItemsTheSame(@NonNull ConsoleWithGames oldItem, @NonNull ConsoleWithGames newItem) {
                return oldItem.getConsole().getId() == newItem.getConsole().getId();
            }

            @Override
            public boolean areContentsTheSame(@NonNull ConsoleWithGames oldItem, @NonNull ConsoleWithGames newItem) {
                return oldItem.equals(newItem);
            }
        };
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof ConsoleWithGames)) return false;
        ConsoleWithGames that = (ConsoleWithGames) o;
        return Objects.equals(getConsole(), that.getConsole()) &&
                Objects.equals(getGames(), that.getGames());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getConsole(), getGames());
    }

}

File: app/src/main/java/br/com/vicentec12/mygames/data/model/ConsoleWithGames.kt
package br.com.vicentec12.mygames.data.model

import android.os.Parcelable
import androidx.recyclerview.widget.DiffUtil
import androidx.room.Embedded
import androidx.room.Relation
import br.com.vicentec12.mygames.data.model.Game.Companion.COLUMN_CONSOLE_ID
import kotlinx.parcelize.Parcelize

@Parcelize
data class ConsoleWithGames(
        @Embedded
        val console: Console,
        @Relation(parentColumn = "id", entityColumn = COLUMN_CONSOLE_ID)
        val games: List<Game>
) : Parcelable {

    companion object {

        val DIFF_UTIL_CALLBACK = object : DiffUtil.ItemCallback<ConsoleWithGames>() {
            override fun areItemsTheSame(oldItem: ConsoleWithGames, newItem: ConsoleWithGames) =
                    oldItem.console.id == newItem.console.id

            override fun areContentsTheSame(oldItem: ConsoleWithGames, newItem: ConsoleWithGames) = oldItem == newItem
        }

    }

}